enhancements/installer/byo-custom-dns.md

```diff
@@ -0,0 +1,540 @@
+IP.
+
+This would work well except for the following 2 reasons:
+1. AWS does not refer to the `LoadBalancerIP` input value before it creates an
+Ingress LB. So, the above solution only would work for Azure and GCP and not
+for AWS.
+
+2. Based on [K8s API documentation] (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#servicespec-v1-core), the `LoadBalancerIP` field would soon be deprecated.
+
+### API Extensions
+
+1. Install config is augmented to allow the customer to specify if a custom
+DNS solution is being used. In addition, the customer would be able to indicate
+the LB type to be used for API and API-Int traffic.
+
+```yaml
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.9.2
+  creationTimestamp: null
+  name: installconfigs.install.openshift.io
+spec:
+  group: install.openshift.io
+  names:
+    kind: InstallConfig
+    listKind: InstallConfigList
+    plural: installconfigs
+    singular: installconfig
+  scope: Namespaced
+  versions:
+  - name: v1
+    schema:
+<snip>
+                    aws:
+                      description: AWS is the configuration used when installing on
+                        AWS.
+                      properties:
+                        userConfiguredDNSLB:
+                          description: "UserConfiguredDNSLB contains all the API and API-Int
+                            LB information. \n This field is used to Enable the use of a
+                            custom DNS solution when the DNS provided by the underlying
+                            cloud platform cannot be used. When Enabled, the user can provide
+                            information about user created API and API-Int LBs using this
+                            field."
+                          properties:
+                            apiIntLBName:
+                              description: ApiIntLBName is the name of the API-Int NLB created
+                                by the user.
+                              type: string
+                            apiLBName:
+                              description: ApiLBName is the name of the API NLB created
+                                by the user.
+                              type: string
+                            userDNS:
+                              default: Disabled
+                              description: UserDNS specifies whether the customer is responsible
+                                for configuring DNS entries for API and API-Int prior to
+                                cluster installation.
+                              enum:
+                              - ""
+                              - Enabled
+                              - Disabled
+                              type: string
+                          type: object
+
+<snip>
+ ```
+Setting `userConfiguredDNS` to `Enabled` and not specifying `apiLBName` and
```

Comment:
Do you mean `userConfiguredDNSLB.userDNS` rather than `userConfiguredDNS` here, right?

Reply:
Correct. Missed updating this part of the proposal.